source/client-side-encryption/client-side-encryption.rst

```diff
@@ -293,6 +295,59 @@ cluster as indicated by the connecting URI. But the optional
 keyVaultClient can be used to route data key queries to a separate
 MongoDB cluster.
 
+If a ``keyVaultClient`` is not passed, and the parent ``MongoClient`` is
+configured with a limited ``maxPoolSize``, the ``keyVaultClient`` is set to an
+internal ``MongoClient``. See `keyVaultClient, metadataClient, and the internal
+MongoClient`_ for configuration behavior.
+
+See `What's the deal with metadataClient, keyVaultClient, and the internal client?`_
+
+keyVaultClient, metadataClient, and the internal MongoClient
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The following pseudo-code describes the configuration behavior for the three ``MongoClient``s:
+
+.. code::
+
+   def getOrCreateInternalClient (client, clientOpts):
+      if client.internalClient != None:
+         return client.internalClient
+      internalClientOpts = copy(clientOpts)
```

Comment:
Does any spec standardize what the `clientOpts` are? Seems like this refers to URI options + other options that can be supplied to MongoClient?
When creating the new client, are we expected to use the host/port supplied to the original MongoClient or are expected instead to create a new topology for the seeds/nodes that are known at the time of internal MongoClient creation? AFAICT, those two could be different so having the spec be more explicit in this regard would be helpful.

Reply:
Good question. AFAIK, no spec standardizes all client options. I added a section below to describe the intended behavior. That the internal client should use the same configuration supplied to the parent `MongoClient`.

I do not think it makes much of an observable difference whether we use the original host information, or if we use the seeds/nodes known at the time of constructing the internal `MongoClient`. But if the internal `MongoClient` is configured during construction, I think it may make implementations simpler to always use the original host info from the URI.

Reply:
I just want the spec to specifically mention what do. I agree that since the internal client will be created at (almost) the same time as the parent MongoClient bearing the `AutoEncryptionOpts`, use of the original host/port vs the currently discovered nodes wouldn't necessarily matter.